patches/src/main/kotlin/app/revanced/patches/spotify/misc/lyrics/ChangeLyricsProviderPatch.kt

```diff
@@ -0,0 +1,148 @@
+package app.revanced.patches.spotify.misc.lyrics
+
+import app.revanced.patcher.extensions.InstructionExtensions.getInstruction
+import app.revanced.patcher.extensions.InstructionExtensions.replaceInstruction
+import app.revanced.patcher.fingerprint
+import app.revanced.patcher.patch.bytecodePatch
+import app.revanced.patcher.patch.stringOption
+import app.revanced.patcher.util.proxy.mutableTypes.MutableMethod
+import app.revanced.patcher.util.proxy.mutableTypes.MutableMethod.Companion.toMutable
+import app.revanced.patches.spotify.misc.extension.IS_SPOTIFY_LEGACY_APP_TARGET
+import app.revanced.util.getReference
+import app.revanced.util.indexOfFirstInstruction
+import app.revanced.util.indexOfFirstInstructionOrThrow
+import com.android.tools.smali.dexlib2.Opcode
+import com.android.tools.smali.dexlib2.builder.instruction.BuilderInstruction35c
+import com.android.tools.smali.dexlib2.iface.instruction.OneRegisterInstruction
+import com.android.tools.smali.dexlib2.iface.reference.MethodReference
+import com.android.tools.smali.dexlib2.immutable.reference.ImmutableMethodReference
+import java.net.InetAddress
+import java.net.URI
+import java.net.URISyntaxException
+import java.net.UnknownHostException
+import java.util.logging.Logger
+
+
+@Suppress("unused")
+val changeLyricsProviderPatch = bytecodePatch(
+    name = "Change lyrics provider",
+    description = "Changes the lyrics provider to a custom one.",
+) {
+    compatibleWith("com.spotify.music")
+
+    val lyricsProviderUrl by stringOption(
+        key = "lyricsProviderUrl",
+        title = "Lyrics provider URL",
+        description = "The URL to a custom lyrics provider.",
+        required = true,
+        default = "lyrics.natanchiodi.fr"
+    ) {
+        val host: String?
+        try {
+            host = URI(it!!).host ?: it
+        } catch (e: URISyntaxException) {
+            return@stringOption false
+        }
+
+        try {
+            InetAddress.getByName(host)
+        } catch (e: UnknownHostException) {
+            Logger.getLogger(this::class.java.name).warning(
+                "Host \"$host\" did not resolve to any domain."
+            )
+        }
+        true
+    }
+
+    execute {
+        if(IS_SPOTIFY_LEGACY_APP_TARGET) {
+            Logger.getLogger(this::class.java.name).severe(
+                "Change lyrics provider patch is not supported for this target version."
+            )
+            return@execute
+        }
+
+
+        //region Create a patched HTTP client for the requested URL
+
+        // Copy the method definition of the HTTP client builder for a valid hostname.
+        val patchedClientMethod : MutableMethod
```

Comment:
Naming consistency is important
```suggestion
        val patchedHttpClientBuilderMethod : MutableMethod
```